semgrep-core/Analyzing/Bloom_annotation.ml

```diff
@@ -53,23 +91,94 @@ module B = Bloom_filter
  * See also Rules_filter.reserved_id and reserved_str and
  * Rules_filter.extract_specific_strings.
 *)
-let extract_strings _any =
-  []
+
+let rec statement_strings stmt =
+  let res = ref [] in
+  let top_level = ref true in
+  let visitor = V.mk_visitor {
+    V.default_visitor with
+    V.kident = (fun (_k, _) (str, _tok) ->
+      if not (special_ident str) then
+        push str res
+    );
+    V.kexpr = (fun (k, _) x ->
+      (match x with
+       (* less: we could extract strings for the other literals too?
+        * atoms, chars, even int?
+       *)
+       | L (String (str, _tok)) ->
+           if not (special_literal str) then push str res
+       | TypedMetavar _ -> ()
+       | _ -> k x
+      )
+    );
+    V.kconstness = (fun (k, _) x ->
+      (match x with
+       | Lit (String (str, _tok)) ->
+           if not (special_literal str) then push str res
+       | _ -> k x
+      )
+    );
+    V.kid_info = (fun (k, _) x -> k x);
```

Comment:
maybe a comment here reminding that the default behavior of kid_info is to not call the continuation? which
is why we redefine here to call the continuation so we can also index propagated constants inside id_constness
(that would not be indexed by the default visitor).